### DISCUSSION TITLE
A Way To Simulate Lightning

### DISCUSSION THREAD

**Discussion by odvigajlo** [niels747/2D-Weather-Sandbox#77] [Ideas]
Lightning is already in todo list, so i want to give Niels some ideas for those.

**Simulation**
Possibly most easiest, realistic, perfomant method to do them is this type of simulation:
So, we already have different types of precipitation, freezing rain/hail, rain, snow
so what i suggest is when rain particle collide with hail or snow particles there's will be chance of summoning lightning bolt out of this place!
And higher speed of collision, higher chance of summoning will be, so powerful updrafts will have more chance to summon lightning bolt, and dying storms will have less and less of them.

**Visuals**
For visual, a pile of sprites can be used, and when needed you can just show random sprite for half of second.

But you can do a more complex work, and do procedural lightning!
I suggest making procedural lightning by:
Making huge "stick" out of place of summoning to random spot on ground, bend it randomly and branch out.

**Mechanics/Gameplay?**
I know you want to make them just visual feature, so only thing i would suggest here:
Fires!
How about lightnings summon small fires?

**Comment by niels747** [comment]
Good ideas, I am still figuring out what the best way to implement lighting is
